src/ip_voter.cpp

```diff
@@ -173,14 +173,31 @@ namespace libtorrent
 		return true;
 	}
 
-	bool external_ip::cast_vote(address const& ip, int source_type, address const& source)
+	external_ip::external_ip(address local4, address global4
+		, address local6, address global6)
 	{
-		return m_vote_group[ip.is_v6()].cast_vote(ip, source_type, source);
+		m_addresses[0][0] = global4;
```

Comment:
would it make sense to aggregate-initialize these?

i.e. ``m_addresses{{global4, global6}, {local4, local6}}`` in the initializer list